docs/using/administering/debugging.rst

```diff
@@ -14,6 +14,35 @@ Ubuntu
 ^^^^^^
 
 XXX This should be documented, see :issue:`1877`.
+# FLOC-2647 Perhaps I should just close 1877 with this branch.
+
+Ubuntu Bug Reporting
+^^^^^^^^^^^^^^^^^^^^
+
+When reporting issues with Flocker on Ubuntu always include copies of all the log files.
+
+The relevant logs can be exported by running the following commands as root:
+
+# FLOC-2647 Perhaps this would be better as a standalone script.
+# Followup issue or in this branch?
+.. prompt:: bash [root@node1]# auto
+
+   SUFFIX="${HOSTNAME}_$(date +%s)"
+   ARCHIVE_NAME="clusterhq_flocker_logs_${SUFFIX}"
+   # Export all logs into a single directory
+   mkdir "${ARCHIVE_NAME}"
+   # Export the raw Eliot log messages from all enabled Flocker services
+   find /var/log/flocker -type f \
+   | xargs --max-args=1 --replace='{}' -- basename {} '.log'  \
+   | while read filename; do
+       gzip < "/var/log/flocker/${filename}.log" > "${ARCHIVE_NAME}/${filename}-${SUFFIX}.log.gz"
+   done
+   # Export the full syslog since last boot with UTC timestamps
+   gzip < /var/log/syslog > "${ARCHIVE_NAME}/all-${SUFFIX}.log.gz"
+   # Create a single archive file
+   tar --create --file "${ARCHIVE_NAME}.tar" "${ARCHIVE_NAME}"
```

Comment:
You should get better compression by tarring first and then gzipping the whole tar file (lots of shared patterns between the different log files).

However, if possible, it would also be good to avoid creating the whole tar file on disk.

This may only be tractable by using a script, or using a `zip` file.